internal/provider/kubernetes/kubernetes.go

```diff
@@ -60,6 +62,17 @@ func New(cfg *rest.Config, svr *config.Server, resources *message.ProviderResour
 	}, nil
 }
 
+// ServeHealthProbes starts serving health probes, including health check and ready check.
```

Comment:
curious why not add this into New() ?

Reply:
Right, we can add to New(), no need to add another func.